sql/mysql/tidb.go

```diff
@@ -8,14 +8,158 @@ import (
 	"context"
 	"fmt"
 	"regexp"
+	"sort"
 	"strings"
 
 	"ariga.io/atlas/sql/internal/sqlx"
+	"ariga.io/atlas/sql/migrate"
 	"ariga.io/atlas/sql/schema"
 )
 
-type tinspect struct {
-	inspect
+type (
+	// tplanApply decorates MySQL planApply.
+	tplanApply struct{ planApply }
+	// tdiff decorates MySQL diff.
+	tdiff struct{ diff }
+	// tinspect decorates MySQL inspect.
+	tinspect struct{ inspect }
+)
+
+// priority computes the priority of each change.
+// TiDB does not support multischema alters (no multiple changes in a single alter statement) and therefore
+// we have to break down each alter. priority helps order the alters so they work.
+// for example - we can't drop a column if it is a foreign key - we have to drop the fk first. this is why DropForeignKey gets a different priority
+func priority(change schema.Change) int {
+	switch c := change.(type) {
+	case *schema.DropIndex:
+		return 0
+	case *schema.DropForeignKey:
+		return 1
+	case *schema.DropAttr:
+		return 2
+	case *schema.DropCheck:
+		return 3
+	case *schema.ModifyIndex:
+		return 4
+	case *schema.ModifyForeignKey:
+		return 5
+	case *schema.DropColumn:
+		return 6
+	case *schema.AddColumn:
+		return 7
+	case *schema.ModifyTable:
+		// each modifyTable should have a single change since we apply `flat` before we sort.
+		return priority(c.Changes[0])
+	case *schema.ModifySchema:
+		// each modifyTable should have a single change since we apply `flat` before we sort.
+		return priority(c.Changes[0])
+	default:
+		return 8
```

Comment:
Why you it can be something like:
```suggestion
	case *schema.ModifyTable:
		// each modifyTable should have a single change since we apply `flat` before we sort.
		return priority(c.Changes[0])
	case *schema.ModifySchema:
		// each modifyTable should have a single change since we apply `flat` before we sort.
		return priority(c.Changes[0])
		case *schema.DropIndex, *schema.DropForeignKey, *schema.ModifyIndex, *schema.ModifyForeignKey, ...:
		return 0
	default:
		1
```